### ISSUE TITLE
Add lang attribute to top html element in Desktop

### ISSUE THREAD

**Issue by 1000TurquoisePogs** [zowe/zlux#394]
See https://www.w3.org/WAI/WCAG21/Techniques/html/H57
Note: the desktop is a multi-app environment, so the lang attribute would indicate user preference, but is no guarantee of the content that will be shown since it is the responsibility of individual apps to have the translations for the language requested. This could mislead accessibility tooling, so feedback from users will be needed to gauge effectiveness.

**Comment by std4lqi** [comment]
A relevant one is [https://www.w3.org/WAI/WCAG21/Techniques/html/H58](https://www.w3.org/WAI/WCAG21/Techniques/html/H58). 

The app could have its own `lang` attribute.

**Comment by nkocsis** [comment]
Is there a target for this fix ?

**Comment by std4lqi** [comment]
It would be great to have it in the next release after 1.9.0. Thanks!

**Comment by grlwholifts** [comment]
Hi @std4lqi @1000TurquoisePogs i would like to take this up. Could you help me get started?

**Comment by 1000TurquoisePogs** [comment]
> A relevant one is https://www.w3.org/WAI/WCAG21/Techniques/html/H58.
>
> The app could have its own lang attribute.

@std4lqi every App has one or more (almost always 1) dom element with the name com-rs-mvd-viewport. This may be able to have a lang attribute, but if it does would accessibility tools actually understand that information?

Do you have tools that you are using to test correct behavior in this regard? Otherwise, I don't know that the changes we make will be effective. We could provide a test branch if you have tools that you want to test with.

**Comment by std4lqi** [comment]
@1000TurquoisePogs Yes, we'll use screen reader for accessibility testing. The screen reader understands DOM. We can test it for sure. Thanks!

This link for reference as well: https://developer.paciellogroup.com/blog/2016/06/using-the-html-lang-attribute/

**Comment by 1000TurquoisePogs** [comment]
OK sounds good. So actually from that reference it seems we do not need an app-specific feature. Apps can just apply the lang tag on each area of their app they think is appropriate. And the Desktop can apply the lang to the HTML tag since that is something the Desktop can guarantee, but the App level must be controlled by the Apps.

**Comment by std4lqi** [comment]
@1000TurquoisePogs Yes, if the app supports other lang, it can define `lang` at app's top element. I agree with you this App level should be controlled by the Apps.

**Comment by 1000TurquoisePogs** [comment]
So, our automated build generated this: https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/zlux/zlux-app-manager/1.9.0-PR-190/zlux-app-manager-1.9.0-20200306.091858.pax

If you would like to test with it, you can extract it into $ROOT_DIR/components/app-server/share, overwriting zlux-app-manager.

I thought this might be easier than having to reinstall all of Zowe to test, but if you would like the full install it can also be generated.

**Comment by std4lqi** [comment]
@1000TurquoisePogs Sorry, it took a while to try out this build. 

According to https://www.w3.org/WAI/WCAG21/Techniques/html/H57.

The objective of this technique is to identify the default language of a document by providing the lang attribute on the html element. 

Zowe Desktop support multi-language depending on browser language setting, as I observed. For example, my browser language setting is Chinese, and logon dialog shows Chinese. Now the lang attribute is "zh-CN" with PR #190. So I think it works well. 

For the App that doesn't support multi-language, it always shows English, which is correct. If this App need be accessibility compliant, it should set its lang attribute as "en" in the proper HTML element, so that screen reader behavior can be predicated.

For the Apps support multi-language, getting the language setting in the same way as Desktop, they should be OK without lang attribute, since Desktop has it in html element already.

From my testing with screen reader, the delay of setting lang attribute doesn't hurt. Thanks again!

**Comment by 1000TurquoisePogs** [comment]
We can't at this time enforce that apps have the lang attribute set, since apps can be made by anyone. In a future version of a conformance program we can require it, such as zowe 2.0
But, for now the code is merged, so I will close this ticket.
